front/src/common/Map/Search/MapSearchStation.tsx

```diff
@@ -35,39 +43,43 @@ export default function MapSearchStation(props) {
 
   // Create playload based on the type of search "name" or "trigram"
 
-  const createPayload = (searchQuery) => ({
+  const createPayload = (searchQuery: (string | string[])[]) => ({
     object: 'operationalpoint',
-    query: ['and', searchQuery, ['=', ['infra_id'], infraID]],
+    query: ['and', searchQuery, ['=', ['infra_id'], infraID]] as SearchQuery,
   });
 
   // Sort on name, and on yardname
-  const orderResults = (results) =>
+  const orderResults = (results: SearchOperationalPointResult[]) =>
     results.sort((a, b) => a.name.localeCompare(b.name) || a.ch.localeCompare(b.ch));
```

Comment:
do the cloning here.

```suggestion
    results.slice().sort((a, b) => a.name.localeCompare(b.name) || a.ch.localeCompare(b.ch));
```

Reply:
Done